packages/server/src/modules/permissions/model.ts

```diff
@@ -0,0 +1,78 @@
+import { db } from '@application';
+import {
+  NormalizedPermission,
+  Permission,
+  TYPE_PERMISSION,
+} from '@chess-tent/models/src';
+import { stepSchema } from '../step/model';
+
+export interface DepopulatedPermission {
+  id: NormalizedPermission['id'];
+  type: NormalizedPermission['type'];
+  holder: NormalizedPermission['holder'];
+  holderType: NormalizedPermission['holderType'];
+  role: NormalizedPermission['role'];
+  object: NormalizedPermission['object'];
+  objectType: NormalizedPermission['objectType'];
+  v?: number;
+}
+
+const permissionSchema = db.createSchema<DepopulatedPermission>(
+  {
+    type: {
+      type: String,
+      ref: TYPE_PERMISSION,
+    } as unknown as DepopulatedPermission['type'],
+    holder: {} as unknown as DepopulatedPermission['holder'],
+    holderType: {} as unknown as DepopulatedPermission['holderType'],
+    role: {} as unknown as DepopulatedPermission['role'],
+    object: {} as unknown as DepopulatedPermission['object'],
+    objectType: {} as unknown as DepopulatedPermission['objectType'],
+    v: {} as unknown as number,
+  },
+  { minimize: false },
+);
+
+stepSchema.index({
```

Comment:
What is this for?

Got a feeling this should be used in `step` module rather then permissions. Seems like a permissions tool that other modules should use?

Reply:
Obi Wan: "This wasn't copy pasted"